src/inference/AdvancedSMC.jl

```diff
@@ -238,18 +238,15 @@ function AbstractMCMC.sample_end!(
     spl::Sampler{<:ParticleInference},
     N::Integer,
     ts::Vector{<:ParticleTransition};
+    resume_from = nothing,
     kwargs...
 )
-    # Set the default for resuming the sampler.
-    resume_from = get(kwargs, :resume_from, nothing)
-
     # Exponentiate the average log evidence.
     # loge = exp(mean([t.le for t in ts]))
     loge = mean(t.le for t in ts)
 
     # If we already had a chain, grab the logevidence.
-    if resume_from !== nothing   # concat samples
-        @assert resume_from isa Chains "resume_from needs to be a Chains object."
```

Comment:
I think we should leave in the `@assert` statement, or at least throw an error when the value of `resume_from` is not a `Chains`, so people know that resume functionality only works for MCMCChains objects.